pkg/chunkenc/memchunk.go

```diff
@@ -474,9 +474,10 @@ func (c *MemChunk) Iterator(ctx context.Context, mintT, maxtT time.Time, directi
 	its := make([]iter.EntryIterator, 0, len(c.blocks)+1)
 
 	for _, b := range c.blocks {
-		if maxt > b.mint && b.maxt > mint {
-			its = append(its, b.iterator(ctx, c.readers, filter))
+		if maxt < b.mint || b.maxt < mint {
+			continue
```

Comment:
Just thinking out loud, clarifying how this addresses https://github.com/grafana/loki/issues/2124.

With the patch: ignore the current block when none of its entries is within the interval `[mint, maxt]`. Otherwise append it for further processing. Or, in other words: if the current block has any entry within the interval `[mint, maxt]` then append it for further processing.

The special case discussed in https://github.com/grafana/loki/issues/2124 corresponds to `b.maxt == mint` in this code section here, and with the old behavior `maxt > b.mint && b.maxt > mint` a block containing an entry within the interval `[mint, maxt]` was erroneously ignored for further processing.